src/modules/cpu/kernel/normalize.hpp

```diff
@@ -0,0 +1,882 @@
+/*
+MIT License
+
+Copyright (c) 2019 - 2024 Advanced Micro Devices, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+// Computes strides
+void compute_strides(Rpp32u *strides, Rpp32u *shape, Rpp32u tensorDim)
+{
+    if (tensorDim > 0)
+    {
+        Rpp32u v = 1;
+        for (Rpp32u i = tensorDim - 1; i > 0; i--)
+        {
+            strides[i] = v;
+            v *= shape[i];
+        }
+        strides[0] = v;
+    }
+}
+
+// Recursive reduction helper function to compute difference of input with mean and squares them up
+template<typename T>
+void compute_diff_square_sum(Rpp32f &output, T *input, Rpp32s inputStride, Rpp32s numElements, Rpp32f mean)
+{
+    if (numElements > 32)
+    {
+        Rpp32s currElements = numElements >> 1;
+        Rpp32f tmp1 = 0, tmp2 = 0;
+
+        // reduce first half and accumulate
+        compute_diff_square_sum(tmp1, input, inputStride, currElements, mean);
+
+        // reduce second half and accumulate
+        compute_diff_square_sum(tmp2, input + currElements * inputStride, inputStride, numElements - currElements, mean);
+
+        tmp1 += tmp2;
+        output += tmp1;
+    }
+    else
+    {
+        // reduce to a temporary
+        Rpp32f tmp = 0;
+        for (Rpp32s i = 0; i < numElements; i++)
+        {
+            Rpp32f curr = (input[i * inputStride] - mean);
+            auto curSq = curr * curr;
+            tmp += curSq;
```

Comment:
why you need tmp? output += (curr * curr)

Reply:
Not in the PR